impl/src/main/java/org/jboss/cdi/tck/tests/extensions/lifecycle/broken/observerMethod/ObserverMethodWithoutNotifyMethodTest.java

```diff
@@ -37,16 +39,37 @@
 @SpecVersion(spec = "cdi", version = "2.0-EDR2")
 public class ObserverMethodWithoutNotifyMethodTest extends AbstractTest {
 
-    @Deployment
+    public static final String CONFIGURATOR = "configuratorRead";
+    public static final String CUSTOM_OM = "customObserverMethod";
+
+    @ArquillianResource
+    Deployer deployer;
+
+    @Deployment(name = CONFIGURATOR, managed = false, testable = false)
     @ShouldThrowException(DefinitionException.class)
-    public static WebArchive createTestArchive() {
+    public static WebArchive createFirstTestArchive() {
         return new WebArchiveBuilder().withTestClass(ObserverMethodWithoutNotifyMethodTest.class)
                 .withClasses(AfterBeanDiscoveryObserver.class, Foo.class, FooObserver.class).withExtension(AfterBeanDiscoveryObserver.class)
-                .build();
+                .withName(CONFIGURATOR + ".war").build();
     }
 
-    @Test
+    @Deployment(name = CUSTOM_OM, managed = false, testable = false)
+    @ShouldThrowException(DefinitionException.class)
+    public static WebArchive createSecondTestArchive() {
+        return new WebArchiveBuilder().withTestClass(ObserverMethodWithoutNotifyMethodTest.class)
+                .withClasses(ExtensionAddingCustomObserver.class, Foo.class, FooObserver.class).withExtension(ExtensionAddingCustomObserver.class)
+                .withName(CUSTOM_OM + ".war").build();
+    }
+
+    @Test(dataProvider = ARQUILLIAN_DATA_PROVIDER)
     @SpecAssertions({ @SpecAssertion(section = AFTER_BEAN_DISCOVERY, id = "ec") })
     public void observerMethodNotOverridingNotifyMethodTreatedAsDefinitionError() {
+        deployer.deploy(CONFIGURATOR);
```

Comment:
I can see compilation problem for `ExtensionAddingCustomObserver` but to your question - well maybe it would be safer to wrap this to try/catch block and do undeploy in the case of test failure (deployment will pass). This would also mean you can remove `@ShouldThrowException`.